job_controller/job_controller.go

```diff
@@ -134,6 +147,22 @@ type JobController struct {
 	// recorder is an event recorder for recording Event resources to the
 	// Kubernetes API.
 	Recorder record.EventRecorder
+
+	// deleteJobHandler deletes the job
+	deleteJobHandler func(job interface{}) error
```

Comment:
Seems like this is just one line. Can we use a generic way to discover the API to call? Then we can get rid of this.

Reply:
I'm not sure how this can be done. 
We plan to write a operator with kube-builder,  with kube-builder, the api for deletion is also different